File: frontend/src/app/entity/project-data.ts
import { HabitCompletionDetails } from "./habit-completion-details";
import { HabitDetails } from "./habit-details";
import { ProjectDetails } from "./project-details";
import { TaskDetails } from "./task-details";

export interface ProjectData {
    project: ProjectDetails;
    tasks: TaskDetails[];
    habits: HabitDetails[];
    //habitCompletions: HabitCompletionDetails[];
}

File: backend/src/main/java/io/github/xpakx/ladder/project/dto/ProjectData.java
package io.github.xpakx.ladder.project.dto;

import io.github.xpakx.ladder.habit.dto.HabitDetails;
import io.github.xpakx.ladder.task.dto.TaskDetails;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class ProjectData {
    private ProjectDetails project;
    private List<TaskDetails> tasks;
    private List<HabitDetails> habits;
}
